docs/pages/blog/turbo-1-10-0.mdx

```diff
@@ -0,0 +1,137 @@
+
+<Date>
+  Thursday, June 1st, 2023
+</Date>
+
+<Authors authors={[
+  "nathanhammond",
+  "tomknickman",
+  "anthonyshew",
+]} />
+
+Turborepo 1.10 introduces new features to improve your local development experience:
+
+- [**Generators**](#generators): Use `turbo gen` to generate source code, including creating new workspaces and copying existing ones.
+- [**Improved Environment Variable Inputs**](#improved-environment-variable-inputs): It's now easier to depend on environment variables with wildcards and `.env` support.
+- [**New Environment Modes**](#environment-modes): You can now specify `--env-mode=strict` to control environment variables that are included in your task hashes.
+
+Update today with
+
+```
+npx @turbo/codemod migrate
+```
+
+## Generators
+
+Turborepo Generators allow you to add new code to your repository in a predictable, structured way.
+Previously, you would need to copy code yourself and manually add dependencies to your new workspace. With Generators,
+we've automated the process of creating new workspaces and integrating them into your repository.
+
+You can [create a blank workspace from scratch](/repo/docs/core-concepts/monorepos/code-generation#add-an-empty-workspace), [copy an existing workspace](/repo/docs/core-concepts/monorepos/code-generation#copy-an-existing-workspace), and even create custom generators
+for more control. Custom generators leverage [Plop](https://plopjs.com/) configurations, a common template definition strategy used in the JavaScript ecosystem.
+
+[Custom generators](/repo/docs/core-concepts/monorepos/code-generation#custom-generators) can also be used to scaffold more granular sets of code than workspaces. You can significantly speed up tasks like:
+
+- [Adding new components to your UI library](https://github.com/vercel/turbo/blob/main/examples/basic/packages/ui/turbo/generators/config.ts)
+- Creating new unit tests
+- Generating a new MDX blog post (In fact, this very post was created with a [Turborepo generator](https://github.com/vercel/turbo/blob/main/docs/turbo/generators/config.ts)!)
+
+Get started with your first generator by running:
+
+```
+turbo gen
+```
+
+For more, check out the [documentation](/repo/docs/core-concepts/monorepos/code-generation).
+
+## Improved Environment Variable Inputs
+
+### Support for `.env` files
+
+Developers and frameworks commonly use `.env` files to load environment variables into their development environments. This can cause problems when running with turbo, as these variables can be [invisible](/repo/docs/core-concepts/caching/environment-variable-inputs#invisible-environment-variables), and are not easily
+included in the hash for your task. This problem could cause an unintended cache hit while working locally.
+
+Turbo now makes this much easier with automatic support for `.env` files.
+
+To ensure that `turbo` includes these variables in your hash, use the `dotEnv` key. For example, here is a configuration for a Next.js application:
```

Comment:
```suggestion
To ensure that `turbo` includes the values of these variables in your hash, use the `dotEnv` key. For example, here is a configuration for a Next.js application:
```

Reply:
I want to try and stay as far away as we can from language that anyone might interpret as "we use this to determine which env vars to depend on" when in fact we just include the file in the hash. If the values are there, we include them. But it's not something we do automatically.